File: py_brains/src/connection.py

from random import random


def random_m1_p1():
    return random() - 0.5


class Connection(object):
    def __init__(self, neuron, weight=random_m1_p1()):
        self._neuron = neuron
        self._weight = weight

    def __str__(self):
        return '%s %s' % (self._neuron, self._weight)

    def get_weight(self):
        return self._weight

    def get_neuron(self):
        return self._neuron

    def alter_weight(self, delta):
        self._weight += delta

    def activate(self):
        return self._neuron.activate() * self._weight

File: sharp_brain/sharp_brain/Neural/Connection.cs
using System;

namespace SharpBrain.Neural
{
    public class Connection
    {
        private readonly Neuron inputNeuron;

        private double weight = 0.0;

        public double Weight
        {
            get { return weight; }
        }

        public double NeuronValue
        {
            get { return inputNeuron.Value; }
        }

        public Connection(Neuron inputNeuron)
        {
            this.inputNeuron = inputNeuron;
            var rand = new Random();
            weight = rand.NextDouble() - 0.5;
        }

        public double Activate()
        {
            return inputNeuron.Activate() * weight;
        }

        public void AdjustWeight(double delta)
        {
            weight += delta;
        }
    }
}
